### PR TITLE
Expand comma separated lists in `global.cylc:[install][symlink dirs]`

### PR DESCRIPTION
**Pull request by wxtim**
From an issue raised by @dpmatthews 

Allow comma separated lists of install targets in  `global.cylc:[install][symlink dirs][__MANY__]`

@oliver-sanders tagged for code review, @dpmatthews for concept review.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -2084,6 +2085,11 @@ def _expand_platforms(self):
             self.sparse['platforms'] = expand_many_section(
```
Comment (oliver-sanders) [inline cylc/flow/cfgspec/globalcfg.py:2085]: Do we not expand platform groups?

Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:2085]: We do not. 

Whilst I guess that it's concievably possible that you might want

```cylc
[platform groups]
  [[*]]
    [[[selection]]]
      method = definition order
  [[platform-one]]
    platforms = a, b
  [[platform-nine-and-three-quarters]]
    platforms = c, d
```

I'm not sure how useful this is for a setting with only two functional options.

**Review by oliver-sanders — APPROVED** [review]

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -587,6 +587,35 @@
     '''
 }
 
+COMMA_SEPARATED_SECTION_NOTE = '''
+
+    .. note::
+
+       This section can be a comma separated list. For example:
+
+       .. code:: cylc
```
Comment (oliver-sanders) [inline cylc/flow/cfgspec/globalcfg.py:596]: We use the Sphinx `code-block` directive in cylc-doc.

File: cylc/flow/network/schema.py [hunk 3]
```diff
@@ -792,6 +792,12 @@ class RuntimeSetting(ObjectType):
     value = String(default_value=None)
 
 
+class TaskEventsSettings(ObjectType):
+    """Key = value setting for a `[runtime][<namespace>]` configuration."""
+    handlers = String(default_value=None)
+    foo = String(default_value=None)
+
+
```
Comment (MetRonnie) [inline cylc/flow/network/schema.py:800]: Commit pushed to wrong branch?

File: cylc/flow/network/schema.py [hunk 4]
```diff
@@ -803,6 +809,8 @@ class Meta:
               of the definition)
             - Job (a record of what was run for a particular submit)
         """)
+    events = Field(TaskEventsSettings)
+
```
Comment (wxtim) [inline cylc/flow/network/schema.py:813]:
```suggestion
```

File: cylc/flow/network/schema.py [hunk 5]
```diff
@@ -792,6 +792,11 @@ class RuntimeSetting(ObjectType):
     value = String(default_value=None)
 
 
+class TaskEventsSettings(ObjectType):
+    """Key = value setting for a `[runtime][<namespace>]` configuration."""
+    handlers = String(default_value=None)
+    foo = String(default_value=None)
+
```
Comment (wxtim) [inline cylc/flow/network/schema.py:799]:
```suggestion
```

**Comment by MetRonnie** [comment]
> Slightly cheekily, I've targeted 3.8.x not master. Although this is a feature, it's for sysadmins not end-users, and I have very high confidence that it won't break anything.

Is this still the case? Currently opened against master and not assigned a milestone. Did Dave request it be added to 8.3.7?

File: cylc/flow/cfgspec/globalcfg.py [hunk 6]
```diff
@@ -1254,6 +1289,8 @@ def default_for(
                  platform configurations.
 
             .. versionadded:: 8.0.0
+
+            {COMMA_SEPARATED_SECTION_NOTE}
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:1293]: I've tried various things to get rid of it (e.g. using textwrap to dedent/reindent the entire thing differently) and can't get this to work.

**Comment by dpmatthews** [comment]
> It's blocking some stuff Dave wants to do - I think it's a bug from his point of view.

No urgency as far as I'm concerned and certainly not a bug.

**Comment by wxtim** [comment]
> > It's blocking some stuff Dave wants to do - I think it's a bug from his point of view.
> 
> No urgency as far as I'm concerned and certainly not a bug.

Changed to 8.4.0 milestone for less cheeky.

File: cylc/flow/cfgspec/globalcfg.py [hunk 7]
```diff
@@ -1209,7 +1243,7 @@ def default_for(
 
         .. versionadded:: 8.0.0
     '''):
-        with Conf('<platform name>', desc='''
+        with Conf('<platform name>', desc=f'''
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:1246]:
```suggestion
        with Conf('<platform name>', desc=dedent('''
```

This is how it's done elsewhere: https://github.com/cylc/cylc-flow/blob/6eb8f61c5b9777cb21ac68eff13aa4f0722f691a/cylc/flow/cfgspec/workflow.py#L1516-L1519

File: cylc/flow/cfgspec/globalcfg.py [hunk 8]
```diff
@@ -587,6 +587,40 @@
...
+
+                [a]
+                    setting = x
+                [b]
+                    setting = x
+                [a]
+                    another_setting = y
+
+            Which will then be combined according to
+            :ref:`the rules for Cylc config syntax<syntax>`.
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:620]:
```suggestion
.. note::

   This section can be a comma separated list.

   .. spoiler:: Example

      For example:

      .. code-block:: cylc

         [a, b]
             setting = x
         [a]
             another_setting = y

      Will become:

      .. code-block:: cylc

         [a]
             setting = x
         [b]
             setting = x
         [a]
             another_setting = y

      Which will then be combined according to
      :ref:`the rules for Cylc config syntax<syntax>`.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 9]
```diff
@@ -1219,7 +1253,7 @@ def default_for(
             Platform names can be regular expressions: If you have a set of
             compute resources such as ``bigmachine1, bigmachine2`` or
             ``desktop0000, .., desktop9999`` one would define platforms with
-            names ``[[bigmachine[12]]]`` and ``[[desktop[0-9]{4}]]``.
+            names ``[[bigmachine[12]]]`` and ``[[desktop[0-9]{{4}}]]``.
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:1256]:
```suggestion
            names ``[[bigmachine[12]]]`` and ``[[desktop[0-9]{4}]]``.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 10]
```diff
@@ -1133,7 +1167,7 @@ def default_for(
         """):
             with Conf('<install target>', desc="""
                 :ref:`Host <Install targets>` on which to create the symlinks.
-            """):
+            """ + COMMA_SEPARATED_SECTION_NOTE):
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:1170]:
```suggestion
            with Conf('<install target>', desc=dedent("""
                :ref:`Host <Install targets>` on which to create the symlinks.
            """) + COMMA_SEPARATED_SECTION_NOTE):
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 11]
```diff
@@ -18,7 +18,7 @@
 import os
 from pathlib import Path
 from sys import stderr
-from textwrap import dedent
+from textwrap import dedent, indent
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:21]:
```suggestion
from textwrap import dedent
```

File: cylc/flow/cfgspec/globalcfg.py (hunk 7 above)
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:1246]:
```suggestion
        with Conf('<platform name>', desc=dedent('''
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 12]
```diff
@@ -1254,7 +1288,8 @@ def default_for(
                  platform configurations.
 
             .. versionadded:: 8.0.0
-        ''') as Platform:
+
+        ''' + indent(COMMA_SEPARATED_SECTION_NOTE, '   ' * 3)) as Platform:
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:1292]:
```suggestion
        ''') + COMMA_SEPARATED_SECTION_NOTE) as Platform:
```

**Review by MetRonnie — CHANGES_REQUESTED** [review]
https://github.com/wxtim/cylc/pull/68

**Comment by hjoliver** [comment]
(Took a quick look, but I think @MetRonnie is about to approve this so I'll leave it be)

**Comment by MetRonnie** [comment]
Sorry, got another one: https://github.com/wxtim/cylc/pull/69

**Review by MetRonnie — APPROVED** [review]

**Comment by hjoliver** [comment]
Two approvals.

**Pull request merged**